octodns/provider/cloudflare.py

```diff
@@ -340,13 +340,21 @@ def _contents_for_MX(self, record):
             }
 
     def _contents_for_SRV(self, record):
-        service, proto = record.name.split('.', 2)
+        n = record.name.split('.')
+        service, proto, subdomain = n[0], n[1], None,
+        if len(n) > 2:
+            service, proto, subdomain = n[0], n[1], '.'.join(n[2:])
```

Comment:
Might be a little more pythonic and easier to read with:

```python
try:
    service, proto, subdomain = record.name.split('.', 2)
    # We have a SRV in a sub-zone
except ValueError:
    # We have a SRV in the zone
    service, proto = record.name.split('.', 1)
    subdomain = None
```

Reply:
I like that much better. I've been doing Go lately so I'm not used to having exceptions :D